### PR TITLE
Remove allocation from ByteSize.TryParse

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/Humanizer/Bytes/ByteSize.cs [hunk 1]
```diff
@@ -386,73 +387,82 @@ public int CompareTo(object? obj)
     public static bool TryParse(string? s, out ByteSize result) =>
         TryParse(s, null, out result);
 
-    public static bool TryParse(string? s, IFormatProvider? formatProvider, out ByteSize result)
+    public static bool TryParse(string? s, IFormatProvider? formatProvider, out ByteSize result) =>
+        TryParse(s.AsSpan(), formatProvider, out result);
+
+    static bool TryParse(CharSpan span, IFormatProvider? formatProvider, out ByteSize result)
```
Comment (hazzik) [inline src/Humanizer/Bytes/ByteSize.cs:393]: Wouldn't we benefit if this become public?

Comment (stephentoub) [reply src/Humanizer/Bytes/ByteSize.cs:393]: From my perspective, yes, which is why I factored it out into its own method. But I didn't want to presume anything about adding public surface area.

Comment (SimonCropp) [reply src/Humanizer/Bytes/ByteSize.cs:393]: yeah. i agree, it should be public

Comment (stephentoub) [reply src/Humanizer/Bytes/ByteSize.cs]: Added the public API

**Pull request merged**